### ISSUE TITLE
ng-model and date-options

### ISSUE THREAD

**Issue by pdemilly** [indrimuska/angular-moment-picker#63]
Would it be possible to add a new directive that takes ng-model as date value and date-options as moment-picker options.

<input type="text" ng-model="mydate" moment-picker-options="myMomentPickerOptions">

this would permit consistent integration with ng-model based forms.

**Comment by indrimuska** [comment]
Hi @pdemilly, thanks for your proposal.

If your goal is to use *Moment Objects* instead of *Strings* for your model, there is an opened issue for that (#60). I am also already working on it in my `return-moment-object` branch.

However, I wonder if you could please provide an example of a *ng-model based forms* in which Angular Moment Picker directive is not working properly (I mean, using a *string* for the model).

Thanks :)

**Comment by pdemilly** [comment]
I created a quick component (angular 1.5) of what I was thinking
`
MYAPP.component ('myMomentPicker', {
	require: {
		ngModelCtrl: '^ngModel',
	},
	bindings: {
		dateOptions: '=',
	},
	template: require('./date.html'),
	controller: /* @ngInject */ function ($scope, $timeout) {
		var $ctrl = this;

		$ctrl.$onInit = function () {
			$ctrl.ngModelCtrl.$render = function model2View () {
				$ctrl.mydate = $ctrl.ngModelCtrl.$viewValue;
			};

			$scope.$watch (function () {
				return $ctrl.mydate;
			}, function (val, oldVal) {
				$timeout(function () {
					$ctrl.ngModelCtrl.$setViewValue ($ctrl.mydate);
				},100);
			});
		};
	},
});
`

date.html
`
<div class="input-group date">
        <input type="text"
                class="form-control"
                ng-model="$ctrl.mydate"
                ng-model-options="{ updateOn: 'blur' }"
                moment-picker="$ctrl.mydate"
                locale="{{::$ctrl.locale || 'en'}}"
                format="{{::$ctrl.dateOptions.format || 'L'}}"
                min-view="{{::$ctrl.dateOptions.minView}}"
                max-view="{{::$ctrl.dateOptions.maxView}}"
                start-view="{{::$ctrl.dateOptions.startView}}"
                autoclose="$ctrl.dateOptions.autoclose || true"
                today="$ctrl.dateOptions.today || true"
                keyboard="$ctrl.dateOptions.keyboard || false"
                years-format="{{::$ctrl.dateOptions.yearsFormat || 'YYYY'}}"
                months-format="{{::$ctrl.dateOptions.monthsFormat || 'MMM'}}"
                days-format="{{::$ctrl.dateOptions.daysFormat || 'D'}}"
        >
        <span class="input-group-addon"><i class="glyphicon glyphicon-th"></i></span>
</div>
`

**Comment by indrimuska** [comment]
Oh I didn't knew Angular's components yet, I need to learn more about them.
Anyway, I don't understand the advantages of such kind of structure.

From the "encapsulation" point of view, using this:
```html
<my-moment-picker ng-model="vm.dob" date-options="{ autoclose: true }" />
```
is not so different than using  this: 
```html
<input moment-picker="vm.dob" autoclose="true">
```

The first one just use a "special directive" to have an icon after the input text. But everyone can create his directive to wrap a special combination of template that includes a moment picker inside, so this shouldn't be a plus.

Instead, it should be the `ngModel` and `ngModelCtrl` combination, but I cannot figure out how I can use them in a way that actually cannot be done. Could you please help me to understand the real gap between your system configuration and mine?

Sorry for the many questions. I really want to enhance this picker and if this feature can add a value to it then this should be part of the next releases. :)

**Comment by pdemilly** [comment]
In forms management it is important to be consistent. In angular v1, a form
(ng-form) follows a standard where each element of the form is modified by
its own ng-model. This permits to have things like form validation,
data-binding and state control working at the form level. Components like
yours need to be compatible if they want to participate .

In my particular case, I use angular-formly to generate my form, and
angular-formly strongly follow the ng-form pattern. With this component (it
could be a directive for angular <1.4) it works

Hope that helps

**Comment by indrimuska** [comment]
Hi @pdemilly, I'm going to release a new major version of the directive that makes use of the `ngModel` directive. Since this issue will be solved by this new feature, I invite you to participate to the conversation in the issue #60 (I think you should look at https://github.com/indrimuska/angular-moment-picker/issues/60#issuecomment-254596657 ).

For that reason, I'm going to close this issue, but feel free to open it again if don't receive an appropriate answer to your questions.
Thanks :)

**Comment by indrimuska** [comment]
Closed by #91.
